pkg/kubectl/cmd/annotate.go

```diff
@@ -104,6 +104,7 @@ func NewCmdAnnotate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 		},
 	}
 	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDiscoveryFlags(cmd)
```

Comment:
> Why do we need the flag, anyway?

In order to find the third party resources, you must make a connection to the apiserver.  That mapping of third party kinds and resources is determined when you get a `RESTMapper`, which is shared by everyone trying to get a mapper: including those bits that allowed the `--local` flag.

Most of the time, we don't want to re-run discovery.  I think that long term we'll want to cache the information and have a way to specify that we want it refreshed.